app/code/Magento/Contact/Model/ContactForm.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Magento\Contact\Model;
+
+use Magento\Framework\Api\AbstractSimpleObject;
+use Magento\Contact\Api\Data\ContactFormInterface;
+
+class ContactForm extends AbstractSimpleObject implements ContactFormInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return $this->_data[self::NAME];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setName($name)
+    {
+        if (trim($name) === '') {
```

Comment:
We should avoid logic in DTO (data objects) models.
If you don't have logic in your DTO then you have the possibility to transfer this data between different systems (contexts)

Also, validation depends on different contexts (in one context it could be valid, in another context the same data will not be valid).
We can extract validation logic to separate process(and this process can be extended or changes)
Example:
https://github.com/magento/async-import/blob/2.3-asynchronous-import-module/app/code/Magento/AsynchronousImportSourceDataRetrieving/Model/RetrieveSourceData.php#L86-L89